src/reducers/notebook-reducer.js

```diff
@@ -46,13 +46,13 @@ function clearHistory(loadedState) {
 function updateAppMessages(state, messageObj) {
   const message = Object.assign({}, messageObj)
   delete message.type
-  const appMessages = state.appMessages.slice()
   if (message.when === undefined) {
     message.when = (new Date()).toString()
```

Comment:
if the `message.when` is required by the message schema, this should never be undefined right? is this check needed?

Reply:
`message.when` is required by the message schema, but that doesn't mean a message will at this point have the correct schema. I believe that only happens after the reducer has completed. I suppose I should just add this into the action as the default.